games/xworld3d/xworld3d_simulator.cpp

```diff
@@ -330,40 +329,38 @@ float X3Simulator::take_action(const StatePacket& actions) {
         size_t action_idx = *(actions.get_buffer("action")->get_id());
         CHECK_LT(action_idx, get_num_actions());
         auto action = legal_actions_[action_idx];
-        if (FLAGS_pause_screen) {
-            switch (key) {
-                case 'w':
-                    action = X3NavAction::MOVE_FORWARD;
-                    break;
-                case 's':
-                    action = X3NavAction::MOVE_BACKWARD;
-                    //                    action = X3NavAction::STOP;
-                    break;
-                case 'a':
-                    action = X3NavAction::MOVE_LEFT;
-                    break;
-                case 'd':
-                    action = X3NavAction::MOVE_RIGHT;
-                    break;
-                case 'q':
-                    action = X3NavAction::TURN_LEFT;
-                    break;
-                case 'e':
-                    action = X3NavAction::TURN_RIGHT;
-                    break;
-                case 'j':
-                    action = X3NavAction::JUMP;
-                    break;
-                case 'c':
-                    action = X3NavAction::COLLECT;
-                    break;
-                case 'z':
-                    bird_view_ = !bird_view_;
-                    break;
-                default:
-                    //                    action = X3NavAction::NOOP;
-                    break;
-            }
+        switch (keyboard_action_) {
```

Comment:
Somewhere in the code you should reset keyboard_action_ = -1 each time step. Otherwise the previous valid keyboard_action will carry over to the remaining time steps.